django/middleware/csrf.py

```diff
@@ -63,21 +101,28 @@ def rotate_token(request):
     """
     request.META.update({
         "CSRF_COOKIE_USED": True,
-        "CSRF_COOKIE": _get_new_csrf_key(),
+        "CSRF_COOKIE": _get_new_csrf_token(),
     })
 
 
 def _sanitize_token(token):
     # Allow only alphanum
-    if len(token) > CSRF_KEY_LENGTH:
-        return _get_new_csrf_key()
```

Comment:
I felt a little uneasy about the original mixing concerns; once it was clear that we need to sanitize both tokens (cookie and post/header), not just one, it became clear that we need to separate new-token-generation from sanitization.